src/vs/workbench/services/textfile/browser/textFileService.ts

```diff
@@ -151,15 +127,15 @@ export abstract class TextFileService implements ITextFileService {
 						return this.confirmBeforeShutdown(); // we still have dirty files around, so confirm normally
 					}
 
-					return this.cleanupBackupsBeforeShutdown();
+					return this.backupService.cleanupBackupsBeforeShutdown(); // all good, no veto
```

Comment:
All of the cleanupBackupsBeforeShutdown should move out into the backup service that itself should listen to lifecycle events.